src/components/spatial/SpatialSubscriber.js

```diff
@@ -168,7 +173,19 @@ export default function SpatialSubscriber(props) {
           },
         );
       }
+      if ((typeof targetX !== 'number' || typeof targetY !== 'number')) {
+        const { initialTargetX, initialTargetY, initialZoom } = getInitialSpatialTargets({
+          width,
+          height,
+          cells,
+          imageLayerLoaders,
+        });
+        setTargetX(initialTargetX);
+        setTargetY(initialTargetY);
+        setZoom(initialZoom);
+      }
     }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [dataset, loaders, autoLayers, imageLayerLoaders,
```

Comment:
Should `cells` be included in this list?

Reply:
Yes!